UnitySDK/Assets/ML-Agents/Scripts/Grpc/RpcCommunicator.cs

```diff
@@ -217,42 +227,53 @@ void SendCommandEvent(CommandProto command)
 
         public void DecideBatch()
         {
-            if (m_CurrentAgents.Values.All(l => l.Count == 0))
+            if (!m_NeedCommunicateThisStep)
             {
                 return;
             }
-            foreach (var brainKey in m_CurrentAgents.Keys)
+            m_NeedCommunicateThisStep = false;
+
+            SendBatchedMessageHelper();
+        }
+
+        /// <summary>
+        /// Sends the observations of one Agent. 
+        /// </summary>
+        /// <param name="brainKey">Batch Key.</param>
+        /// <param name="agent">Agent info.</param>
+        public void PutObservations(string brainKey, AgentInfo info, List<ISensor> sensors, Action<AgentAction> action)
+        {
+            using (TimerStack.Instance.Scoped("AgentInfo.ToProto"))
```

Comment:
Change the name of the timer section and/or move it to just be around info.ToAgentInfoProto().

I think it's worthwhile to have a timer around Agent.GenerateSensorData, either here or in the method itself.

Reply:
ok, done